platform/mellanox/mlnx-platform-api/sonic_platform/modules_mgmt.py

```diff
@@ -458,6 +459,28 @@ def power_on_module(self, port, module_sm_obj, dynamic=False):
                 return STATE_HW_NOT_PRESENT
         return STATE_NOT_POWERED
 
+    def is_supported_for_software_control(self, xcvr_api):
+        return isinstance(xcvr_api, cmis.CmisApi) or isinstance(xcvr_api, sff8636.Sff8636Api) or isinstance(xcvr_api, sff8436.Sff8436Api)
+
+    def update_frequency(self, port, xcvr_api):
+        # first read the frequency support - if it's 1 then continue, if it's 0 no need to do anything
+        module_fd_freq_support_path = SYSFS_INDEPENDENT_FD_FREQ_SUPPORT.format(port)
+        val_int = utils.read_int_from_file(module_fd_freq_support_path)
+        if 1 == val_int:
+            # read the module maximum supported clock of Management Comm Interface (MCI) from module EEPROM.
+            # from byte 2 bits 3-2:
+            # 00b means module supports up to 400KHz
+            # 01b means module supports up to 1MHz
+            logger.log_debug(f"check_module_type reading mci max frequency for port {port}")
+            read_mci = xcvr_api.xcvr_eeprom.read_raw(2, 1)
```

Comment:
Could you please confirm with Eddy/Andy R that SFF8636/SFF8436 also use the same offset as CMIS? As far as I know, CMIS EEPROM memory page is much different than SFF8436/SFF8636. And I did not find anything about "Management Comm Interface" in SFF8636/SFF8436 standard. If possible, could you please point me to the chapter of SFF8636/SFF8436 standard which is related to this part?

Reply:
Checked it with Andy. Frequency for SFF modules is always 400KHz. Updated code accordingly.